docs/manual/interning-checker.tex

```diff
@@ -148,40 +158,95 @@
+\subsectionAndLabel{The intern() methods}{interning-intern-methods}
+Some interned classes use an \<intern()> method to look up the interned version of
+the object. These methods must be annotated with the declaration annotation
+\<@InternMethod>. This allows the checker to verify that a newly created object
+is immediately interned and therefore not issue an interned object creation
+error.
 
+\begin{Verbatim}
+new InternedClass().intern() // no error
+\end{Verbatim}
 
-\subsectionAndLabel{Implicit qualifiers}{interning-implicit-qualifiers}
+Because an \<intern> method is expected to be called on uninterned objects, the
+type of \<this> in \<intern> is implicitly \<@UnknownInterned>. This will cause an
+error if \<this> is used someplace where an interned object is expected.  Some
+of these warnings will be false positives that should be suppressed by the
+user.
 
-The Interning Checker
-adds implicit qualifiers, reducing the number of annotations that must
-appear in your code (see Section~\ref{effective-qualifier}).
-For example, String literals and the \<null> literal are always considered interned, and
-object creation expressions (using \code{new}) are never considered
-\refqualclass{checker/interning/qual}{Interned} unless they are annotated as such, as in
+\begin{Verbatim}
+@InternMethod
+public InternedClass intern() {
+  // Type of "this" inside an @InternMethod is @UnknownInterned
+  @Interned InternedClass that = this; // error
+
+  if (!pool.contains(this)) {
+    @SuppressWarning("interning:assignment.type.incompatible")
+    @Interned InternedClass internedThis = this;
+    pool.add(internedThis);
+  }
+  return pool.get(this);
+}
+\end{Verbatim}
+
+Some interned classes do not use an intern method to ensure that every object
+of that class is interned.  For these classes, the user will have to manually
+inspect every constructor invocation and suppress the ``interned.object.creation''
+error.
+
+If every invocation of a constructor is guaranteed to be interned, then the
+user should annotate the constructor result with \<@Interned> and suppress a
+warning at the constructor.
 
-%BEGIN LATEX
-\begin{smaller}
-%END LATEX
 \begin{Verbatim}
-@Interned Double internedDoubleZero = new @Interned Double(0); // canonical representation for Double zero
+@Interned class AnotherInternedClass {
+  // manually verified that all constructor invocations used such that all
+  // new objects are interned
+  @SuppressWarnings("super.invocation.invalid")
+  @Interned AnotherInternedClass() {}
+}
 \end{Verbatim}
-%BEGIN LATEX
-\end{smaller}
-%END LATEX
+
+
+\subsectionAndLabel{Default qualifiers and qualifiers for literals}{interning-implicit-qualifiers}
+
+The Interning Checker
+adds qualifiers to unannotated types, reducing the number of annotations that must
+appear in your code (see Section~\ref{effective-qualifier}).
+For example, String literals and the \<null> literal are always considered interned.
```

Comment:
Are there any others besides String literals and null? I'd give the complete list here, since otherwise this feels vague.